Extend docs on compositional coupling with deadlock example (#498)
precice/precice.github.io: pages/docs/configuration/configuration-coupling-multi.md

```xml
<coupling-scheme:parallel-explicit>
  <participants first="A" second="B"/>
  ...
</coupling-scheme:parallel-explicit>
<coupling-scheme:parallel-explicit>
  <participants first="C" second="A"/>
  ...
</coupling-scheme:parallel-explicit>
```

> For this example, all three participants are executed in parallel to one another, whereas `A` exchanges data with `B` and `C`, but not the latter two with each other.

Reviewer: ```suggestion
For this example, all three participants are executed in parallel to one another, whereas `A` exchanges data with `B` and `C`, but `B` and `C` do not exchange data with each other.
```
I know it says the same, but I think it's better to make it even more clear. 

Maybe it will be nice to add a comment that "first" and "second" doesn't matter?

Still, even with only explicit schemes, you can do very strange combinations. Some of them will not work, but run into a deadlock. To better understand this, let us modify the example slightly: We add an interaction between `B` and `C` and switch to serial coupling schemes everywhere.

```xml
<coupling-scheme:serial-explicit>
    <participants first="A" second="B" />
    ...
    <exchange data="AB-Data" mesh="..." from="A" to="B" />
  </coupling-scheme:serial-explicit>

  <coupling-scheme:serial-explicit>
    <participants first="B" second="C" />
    ...
    <exchange data="BC-Data" mesh="..." from="B" to="C" />
  </coupling-scheme:serial-explicit>

  <coupling-scheme:serial-explicit>
    <participants first="C" second="A" />
    ...
    <exchange data="CA-Data" mesh="..." from="C" to="A" />
  </coupling-scheme:serial-explicit>
```

The mesh over which the data is communicated plays no role. We get a circular dependency:

```mermaid
graph LR;
    A -- first to second --> B;
    B -- first to second --> C;

>     C -- first to second --> A;

Author: ```suggestion
```txt
A -- first to second --> B;
B -- first to second --> C;
C -- first to second --> A;
```

> ```

Reviewer 2: Does not work at the moment: https://github.com/precice/precice.github.io/issues/552
